rkyv/src/util/aligned_vec.rs

```diff
@@ -11,6 +18,25 @@ use core::{
 use std::{alloc, io};
 
 /// A vector of bytes that aligns its memory to 16 bytes.
+///
+/// The alignment also applies to [`ArchivedAlignedVec`], which is useful for aligning opaque bytes inside of an archived data
+/// type.
+///
+/// ```
+/// # use rkyv::{AlignedVec, Archive, Serialize};
+/// #
+/// #[derive(Archive, Serialize)]
+/// struct HasAlignedBytes {
+///     pub bytes: AlignedVec,
+/// }
+///
+/// let mut bytes = AlignedVec::new();
+/// bytes.extend_from_slice(&[1, 2, 3]);
+/// let ser_foo = rkyv::to_bytes::<_, 0>(&HasAlignedBytes { bytes }).unwrap();
+/// let arch_foo = unsafe { rkyv::archived_root::<HasAlignedBytes>(&ser_foo) };
+/// assert_eq!(arch_foo.bytes.as_slice(), &[1, 2, 3]);
+/// assert_eq!(arch_foo.bytes.as_ptr().align_offset(16), 0);
```

Comment:
This actually succeeds even without the call to `Serializer::align`. @djkoloski Do you have an idea for how to force the re-alignment? I tried putting another field on the struct, but it didn't make a difference.

Reply:
Hmm, we could manually construct a serializer with an intentionally mis-aligned buffer. This would be something like a `BufferSerializer` around `&mut [u8]` pulled out of `AlignedBytes<256>`. Then we would write a single byte so the alignment would be 1 and serializing the `AlignedVec` would require 15 bytes of padding. Asserting before / after should do the trick.